examples/functions/basic/ClassInFrame.xml

```diff
@@ -0,0 +1,134 @@
+	<PublicationTimestamp>2001-12-17T09:30:47.0Z</PublicationTimestamp>
+	<ParticipantRef>SYS001</ParticipantRef>
+	<dataObjects>
+		<CompositeFrame version="1" created="2009-12-02T00:00:00Z" id="RATP:Neptune:CompositeFrame:100110107">
+			<Name>NEPTUNE Mapping Frame</Name>
+			<TypeOfFrameRef version="01" ref="Neptune:TypeOfFrame:CompositeFrame"/>
+			<codespaces>
+				<Codespace id="ratp_pivi_data">
+					<Xmlns>RATP_PIVI</Xmlns>
+					<XmlnsUrl>http://www.ratp.fr/ratp</XmlnsUrl>
+					<Description>RATP datas</Description>
+				</Codespace>
+			</codespaces>
+			<FrameDefaults>
+				<DefaultCodespaceRef ref="ratp_pivi_data"/>
+				<DefaultLocale>
+					<TimeZoneOffset>-1</TimeZoneOffset>
+					<SummerTimeZoneOffset>-2</SummerTimeZoneOffset>
+					<DefaultLanguage>fr</DefaultLanguage>
+				</DefaultLocale>
+				<DefaultLocationSystem>EPSG:4326</DefaultLocationSystem>
+			</FrameDefaults>
+			<frames>
+				<ResourceFrame version="any" id="Neptune:ResourceFrame:FrameProfile01">
+					<Name>RATP--Neptune-Line Profile </Name>
+					<codespaces>
+						<Codespace id="neptune_data">
+							<Xmlns>Neptune</Xmlns>
+							<XmlnsUrl>http://www.Neptune.fr/Neptune</XmlnsUrl>
+							<Description>Neptune data </Description>
+						</Codespace>
+					</codespaces>
+					<FrameDefaults>
+						<DefaultCodespaceRef ref="neptune_data"/>
+					</FrameDefaults>
+					<typesOfValue>
+						<TypeOfValidity version="01" id="Neptune:TypeOfValidity:WeeklyPlanned">
+							<Name>Weekley export of Neptune  Planned data</Name>
+							<Periodicity>P7D</Periodicity>
+							<Nature>planned</Nature>
+						</TypeOfValidity>
+						<TypeOfFrame version="01" id="Neptune:TypeOfFrame:CompositeFrame">
+							<Name>Neptune Profile:  Composite Frame </Name>
+							<Description>RATP--Neptune-Line-xxxxxx  frames will be composite frame containing Service, Timetable (and Service Calendar) and Resource frames with all the usual NEPTUNE attributes filled in, for a given line xxxxx.</Description>
+							<TypeOfValidityRef version="01" ref="Neptune:TypeOfValidity:WeeklyPlanned"/>
+							<FrameClassRef nameOfClass="CompositeFrame"/>
+							<classes>
+								<ClassInFrameRef nameOfClass="AvailabilityCondition" classRefType="memberReferences"/>
+							</classes>
+							<includes>
+								<TypeOfFrame version="01" id="Neptune:TypeOfFrame:TimeTableFrame">
+									<Name>Neptune Profile:ServiceCalendar  Frame </Name>
+									<FrameClassRef nameOfClass="TimeTableFrame"/>
+									<classes>
+										<ClassInFrameRef classRefType="members" nameOfClass="TimeTableFrame">
+											<MemberRef nameOfClass="passingTimes" mandatory="required"/>
+											<MemberRef nameOfClass="calls" mandatory="notAllowed"/>
```

Comment:
```suggestion
											<MemberRef nameOfClass="calls" use="prohibited"/>
```
I would go for the semantics of XML Schema, maybe even use the same types?

Reply:
@Aurige  What do you think? Do we have code that is enforcing  ClassInFrameRef already? I think not, but if yes, can you point me to it?